code/parachain/runtime/dali/src/lib.rs

```diff
@@ -1455,13 +1457,17 @@ impl_runtime_apis! {
 		}
 	}
 
-	impl assets_runtime_api::AssetsRuntimeApi<Block, CurrencyId, AccountId, Balance> for Runtime {
+	impl assets_runtime_api::AssetsRuntimeApi<Block, CurrencyId, AccountId, Balance, ForeignAssetId> for Runtime {
 		fn balance_of(SafeRpcWrapper(asset_id): SafeRpcWrapper<CurrencyId>, account_id: AccountId) -> SafeRpcWrapper<Balance> /* Balance */ {
 			SafeRpcWrapper(<Assets as fungibles::Inspect::<AccountId>>::balance(asset_id, &account_id))
 		}
 
-		fn list_assets() -> Vec<Asset> {
-			CurrencyId::list_assets()
+		fn list_assets() -> Vec<Asset<ForeignAssetId>> {
+			let mut assets = CurrencyId::list_assets();
+			let mut foreign_assets = assets_registry::Pallet::<Runtime>::get_foreign_assets_list();
+			assets.append(&mut foreign_assets);
+
+			assets
```

Comment:
This needs to take into account assets that may be in both

Reply:
(otherwise there will be duplicated assets)

Reply:
Makes sense. But how can we know? Local ids would be different anyway no?

Reply:
@dzmitry-lahoda left a comment relating to this as well, I think he may have more insight into how to properly handle this than I do

Reply:
@benluelo as far as I understand, there will be no duplicated assets. Registering an asset requires sudo, and nobody will register one that is already hardcoded.

Reply:
these are hardcode in PR for XCM. I can fix this RPC if you will at least hardcode KSM = (1,Here).into()